### ISSUE TITLE
Bookmarks to save also line position within file

### ISSUE THREAD

**Issue by arbitrary-dev** [scrooloose/nerdtree#1041]
#### Description

It'd be nice to have `:Bookmark [name]` available from within the file buffers, which will also bookmark current line position, where the cursor is.

**Comment by PhilRunninger** [comment]
I'll give you a solution for the second part of the request. The last known position in every file you edit is stored in the `.viminfo` file, not by any plugin, but by vim itself. You can add just a bit of code to your `vimrc` to make use of this information. This will bring you back to where you were the last time you edited the file.

```
" When editing a file, always jump to its last known cursor position.
augroup jumpToPreviousLocation
    autocmd!
    autocmd BufReadPost * if line("'\"") > 1 && line("'\"") <= line("$") | execute "normal! g`\"" | endif
augroup END
```

The `:Bookmark` command requires NERDTree to be open, so there's no point to making it available outside of the NERDTree window.

**Comment by PhilRunninger** [comment]
@arbitrary-dev I just thought of a way you could do what you're asking with the `:Bookmark` command. You could map a key to find the file in NERDTree and then call the `:Bookmark` command on it. It would look like this, but of course, you can change the key being mapped.
```vim
nnoremap <leader>B :NERDTreeFind<CR>:Bookmark<space>
```
--or--
```vim
nnoremap <leader>B :NERDTreeFind<CR>:Bookmark<CR>
```
The first one waits for you to give a name to the bookmark, while the second on uses the default name without asking.

**Comment by arbitrary-dev** [comment]
So regarding NERDTree, there's no way to have 2 bookmarks on the same file, but on separate lines? :-(

**Comment by PhilRunninger** [comment]
That's right. NERDTree allows you to have multiple bookmarks, but it doesn't store position with them, and the `.viminfo` file stores only one location per file.

**Comment by arbitrary-dev** [comment]
Isn't impossible to implement this within NERDTree, why have you closed this Feature Request?

**Comment by PhilRunninger** [comment]
Here is why I closed the issue:

1. As a file grows and shrinks, a bookmark that includes a line number becomes less and less effective.
1. Should a user also implement the autocmd I presented above, there would be confusion over which location should be used.
1. I am not obliged to implement every feature request that is submitted. The creator of NERDTree and anyone who has worked on it extensively freely admit that the plugin is a big pile of bloat, and adding features should only be done with careful forethought.
